File: generator/src/main/java/ch/bailu/gtk/converter/RelativeNamespaceType.java
package ch.bailu.gtk.converter;

public class RelativeNamespaceType {

    private final NamespaceType type;
    private final boolean hasCurrentNamespace;



    public RelativeNamespaceType(String currentNamespace, String typeName) {
        this(currentNamespace, new NamespaceType(currentNamespace, typeName));
    }

    public RelativeNamespaceType(String currentNamespace, NamespaceType namespaceType) {
        type = namespaceType;
        hasCurrentNamespace = type.getNamespace().equals(currentNamespace);

    }

    public boolean isValid() {
        return type.isValid();
    }

    public boolean hasCurrentNamespace() {
        return hasCurrentNamespace;
    }

    public String getNamespace() {
        return type.getNamespace();
    }

    public String getName() {
        return type.getName();
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/converter/RelativeNamespaceType.kt
package ch.bailu.gtk.converter

class RelativeNamespaceType(currentNamespace: String, namespaceType: NamespaceType) {
    private val type: NamespaceType = namespaceType
    private val hasCurrentNamespace = namespaceType.namespace == currentNamespace


    constructor(currentNamespace: String, typeName: String)
            : this(currentNamespace, NamespaceType(currentNamespace, typeName))


    fun isValid(): Boolean {
        return type.isValid()
    }

    fun hasCurrentNamespace(): Boolean {
        return hasCurrentNamespace
    }

    val namespace: String
        get() = type.namespace

    val name: String
        get() = type.name
}